nifi-nar-bundles/nifi-asn1-bundle/nifi-asn1-services/src/main/java/org/apache/nifi/jasn1/preprocess/AsnPreprocessorEngine.java

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.jasn1.preprocess;
+
+import org.apache.nifi.jasn1.preprocess.preprocessors.ConstraintAsnPreprocessor;
+import org.apache.nifi.jasn1.preprocess.preprocessors.HuggingCommentAsnPreprocessor;
+import org.apache.nifi.jasn1.preprocess.preprocessors.VersionBracketAsnPreprocessor;
+import org.apache.nifi.logging.ComponentLog;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+public class AsnPreprocessorEngine {
+    public static final String COMMA = "\\s*,\\s*";
+
+    private static final List<AsnPreprocessor> PREPROCESSORS = Arrays.asList(
+            new HuggingCommentAsnPreprocessor(),
+            new VersionBracketAsnPreprocessor(),
+            new ConstraintAsnPreprocessor()
+    );
+
+    public String preprocess(
+            ComponentLog componentLog,
+            String asnFilesString,
+            String outputDirectory
+    ) {
+        final String[] inputFiles = asnFilesString.split(COMMA);
+
+        final StringJoiner preprocessedInputFiles = new StringJoiner(",");
+
+        for (String inputFile : inputFiles) {
+            final Path inputFilePath = Paths.get(inputFile);
+            final Path fileName = inputFilePath.getFileName();
+
+            final List<String> lines = readAsnLines(componentLog, inputFile, inputFilePath);
+
+            final List<String> preprocessedLines = preprocessAsn(lines);
+
+            final String preprocessedAsn = preprocessedLines
+                    .stream()
+                    .collect(Collectors.joining(System.lineSeparator()));
+
+            final Path preprocessedAsnPath = Paths.get(outputDirectory, fileName.toString());
+            preprocessedInputFiles.add(preprocessedAsnPath.toString());
+
+            writePreprocessedAsn(componentLog, preprocessedAsn, preprocessedAsnPath);
+        }
+
+        return preprocessedInputFiles.toString();
+    }
+
+    List<String> preprocessAsn(List<String> lines) {
+        List<String> preprocessedAsn = lines;
+
+        for (AsnPreprocessor preprocessor : getPreprocessors()) {
+            preprocessedAsn = preprocessor.preprocessAsn(preprocessedAsn);
+        }
+
+        return preprocessedAsn;
+    }
+
+    List<String> readAsnLines(ComponentLog componentLog, String inputFile, Path inputFilePath) {
+        List<String> lines;
+        try {
+            lines = Files.readAllLines(inputFilePath);
+        } catch (IOException e) {
+            componentLog.error("Couldn't read {}", inputFile, e);
+            throw new UncheckedIOException(e);
```

Comment:
Instead of logging the error, this exception should contain the message. The caller should catch and log the error.
```suggestion
            throw new UncheckedIOException(String.format("Read ASN.1 Schema failed [%s]", inputFile), e);
```